document-store/src/main/java/org/hypertrace/core/documentstore/parser/SelectingExpressionParser.java

```diff
@@ -0,0 +1,16 @@
+package org.hypertrace.core.documentstore.parser;
+
+import org.hypertrace.core.documentstore.expression.impl.AggregateExpression;
+import org.hypertrace.core.documentstore.expression.impl.ConstantExpression;
+import org.hypertrace.core.documentstore.expression.impl.FunctionExpression;
+import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
+
+public interface SelectingExpressionParser {
+  Object parse(AggregateExpression expression);
```

Comment:
Like `SortingExpressionParser`, shouldn't the rest of the interfaces of parse be taking `final` argument?

Reply:
Yup. Good catch. I added the missing `final`s